agent/tool-scripts/bpftrace

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+# -*- mode: shell-script; indent-tabs-mode: t; sh-basic-offset: 8; sh-indentation: 8; sh-indent-for-case-alt: + -*-
+# The following script runs a bpftrace program with the program supplied as argument in pbench-register-tool
+# pbench-register-tool --name=bpftrace -- --script=sample_prog.bt
+
+script_path=$(dirname $0)
+script_name=$(basename $0)
+pbench_bin="$(
+	cd ${script_path}/..
+	/bin/pwd
+)"
+
+# source the base script
+. "$pbench_bin"/base
+
+# Perftool scripts must provide the following functions
+# 1) Install the tool
+# 2) Start data collection
+# 3) Stop data collection
+# 4) post-process the data
+
+# Defaults
+tool=bpftrace
+tool_bin=/usr/bin/$tool
+group=default
+dir="/tmp"
+
+opts=$(getopt -q -o idp --longoptions "dir:,script:,group:,iteration:,start,stop,install,postprocess" -n "getopt.sh" -- "$@")
+if [ $? -ne 0 ]; then
+	printf "\n"
+	printf "$script_name: you specified an invalid option\n\n"
+	printf "The following options are available: \n\n"
+	printf -- "\t--install,			install this perf tool\n"
+	printf "\n"
+	printf -- "\t--start|stop|post-process	start/stop/post-process the data collection\n"
+	printf -- "\t--iteration=int 		the iteration (required)\n"
+	printf -- "\t--group=str		the perftool group (required)\n"
+	printf -- "\t--dir=str			directory to store data collection (required)\n"
+	printf -- "\t--script=str		path to bpftrace script\n"
+	exit 1
+fi
+eval set -- "$opts"
+while true; do
+	case "$1" in
+	--install)
+		mode="install"
+		shift
+		;;
+	--start)
+		mode="start"
+		shift
+		;;
+	--stop)
+		mode="stop"
+		shift
+		;;
+	--postprocess)
+		mode="postprocess"
+		shift
+		;;
+	-d | --dir)
+		shift
+		if [ -n "$1" ]; then
+			dir="$1"
+			shift
+		fi
+		;;
+	-g | --group)
+		shift
+		if [ -n "$1" ]; then
+			group="$1"
+			shift
+		fi
+		;;
+	-i | --iteration)
+		shift
+		if [ -n "$1" ]; then
+			iteration="$1"
+			shift
+		fi
+		;;
+	--script)
+		shift
+		if [ -n "$1" ]; then
+			script=$(echo $1 | sed 's/^"\(.*\)"$/\1/')
+			shift
+		else
+			echo "you did not provide a script"
+		fi
+		;;
+	--)
+		shift
+		break
+		;;
+	esac
+done
+
+# Check for location of bpftrace binary
+if [[ ! -e "/usr/bin/$tool" ]]; then
+	if [[ ! -e "/usr/bin/$tool" ]]; then
+		error_log "$tool_bin not found, exiting"
+		exit 1
+	else
+		tool_bin=/usr/local/bin/$tool
+	fi
+fi
+tool_dir="$dir/tools-$group"
+tool_output_dir=$tool_dir/$tool # all tools keep data in their tool specific dir
+tool_cmd_file="$tool_output_dir/$tool.cmd"
+tool_pid_file=$pbench_tmp/$group.$iteration.$tool.pid
+tool_stdout_file=$tool_output_dir/$tool-stdout.txt
+tool_stderr_file=$tool_output_dir/$tool-stderr.txt
+tool_cmd="$tool_bin $script"
+case "$mode" in
+install)
+	if [[ ! -e $tool_bin ]]; then
```

Comment:
Can we use `check_install_rpm bpftrace` instead?  See other tool files for an example.